docs/modules/ROOT/pages/api/nonFungibleToken.adoc

```diff
@@ -0,0 +1,385 @@
+```
+
+[.contract]
+[[NonFungibleToken]]
+=== `++NonFungibleToken++` link:https://github.com/OpenZeppelin/compact-contracts/blob/main/contracts/nonFungibleToken/src/NonFungibleToken.compact[{github-icon},role=heading-link]
+
+[.contract-index]
+.Circuits
+--
+[.sub-index#NonFungibleTokenModule]
+* xref:#NonFungibleTokenModule-initialize[`++initialize(name_, symbol_)++`]
+* xref:#NonFungibleTokenModule-balanceOf[`++balanceOf(owner)++`]
+* xref:#NonFungibleTokenModule-ownerOf[`++ownerOf(tokenId)++`]
+* xref:#NonFungibleTokenModule-name[`++name()++`]
+* xref:#NonFungibleTokenModule-symbol[`++symbol()++`]
+* xref:#NonFungibleTokenModule-tokenURI[`++tokenURI(tokenId)++`]
+* xref:#NonFungibleTokenModule-_setTokenURI[`++_setTokenURI(tokenId, tokenURI)++`]
+* xref:#NonFungibleTokenModule-approve[`++approve(to, tokenId)++`]
+* xref:#NonFungibleTokenModule-getApproved[`++getApproved(tokenId)++`]
+* xref:#NonFungibleTokenModule-setApprovalForAll[`++setApprovalForAll(operator, approved)++`]
+* xref:#NonFungibleTokenModule-isApprovalForAll[`++isApprovalForAll(owner, operator)++`]
+* xref:#NonFungibleTokenModule-transferFrom[`++transferFrom(from, to, tokenId)++`]
+* xref:#NonFungibleTokenModule-_unsafeTransferFrom[`++_unsafeTransferFrom(from, to, tokenId)++`]
+* xref:#NonFungibleTokenModule-_ownerOf[`++_ownerOf(tokenId)++`]
+* xref:#NonFungibleTokenModule-_getApproved[`++_getApproved(tokenId)++`]
+* xref:#NonFungibleTokenModule-_isAuthorized[`++_isAuthorized(owner, spender, tokenId)++`]
+* xref:#NonFungibleTokenModule-_checkAuthorized[`++_checkAuthorized(owner, spender, tokenId)++`]
+* xref:#NonFungibleTokenModule-_update[`++_update(to, tokenId, auth)++`]
+* xref:#NonFungibleTokenModule-_mint[`++_mint(to, tokenId)++`]
+* xref:#NonFungibleTokenModule-_unsafeMint[`++_unsafeMint(to, tokenId)++`]
+* xref:#NonFungibleTokenModule-_burn[`++_burn(tokenId)++`]
+* xref:#NonFungibleTokenModule-_transfer[`++_transfer(from, to, tokenId)++`]
+* xref:#NonFungibleTokenModule-_unsafeTransfer[`++_unsafeTransfer(from, to, tokenId)++`]
+* xref:#NonFungibleTokenModule-_approve[`++_approve(to, tokenId, auth)++`]
+* xref:#NonFungibleTokenModule-_setApprovalForAll[`++_setApprovalForAll(owner, operator, approved)++`]
+* xref:#NonFungibleTokenModule-_requireOwned[`++_requireOwned(tokenId)++`]
+--
+
+[.contract-item]
+[[NonFungibleTokenModule-initialize]]
+==== `[.contract-item-name]#++initialize++#++(name_: Opaque<"string">, symbol_: Opaque<"string">) → []++` [.item-kind]#circuit#
+
+Initializes the contract by setting the name and symbol.
+
+This MUST be called in the implementing contract's constructor.
+Failure to do so can lead to an irreparable contract.
+
+Requirements:
+
+- Contract is not initialized.
+
+[.contract-item]
+[[NonFungibleTokenModule-balanceOf]]
+==== `[.contract-item-name]#++balanceOf++#++(owner: Either<ZswapCoinPublicKey, ContractAddress>) → Uint<128>++` [.item-kind]#circuit#
+
+Returns the number of tokens in `owner`'s account.
```

Comment:
```suggestion
Returns the number of tokens in ``owner``'s account.
```
Double the backticks when there's an apostrophe otherwise it won't highlight